cmd/gardener-resource-manager/app/app.go

```diff
@@ -194,7 +206,22 @@ func NewResourceManagerCommand() *cobra.Command {
 		rootCAControllerOpts,
 		projectedTokenMountWebhookOpts,
 	)
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
 	verflag.AddFlags(cmd.Flags())
 
 	return cmd
 }
+
+type Options struct {
+	// EnableProfiling enables profiling via web interface host:port/debug/pprof/.
+	EnableProfiling bool
+	// EnableContentionProfiling enable lock contention profiling, if profiling is enabled
+	EnableContentionProfiling bool
+}
+
+// AddFlags adds gardener-resource-manager's flags to the specified FlagSet.
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&o.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
+	fs.BoolVar(&o.EnableContentionProfiling, "contention-profiling", false, "Enable lock contention profiling, if profiling is enabled")
+}
```

Comment:
Shouldn't these flags rather be added to 

https://github.com/gardener/gardener/blob/df2659c5615a7e42aa5823bfdf337253a05d4cad/pkg/resourcemanager/cmd/manager.go#L65-L79

?
Most of the common flags seem to be maintained there.

Reply:
These are the flags specific to the manager, Manager option has no flags related to profiling.

Reply:
While I don't think the new flags should necessarily be added to `ManagerOptions`, I do think that for consistency reasons they should be handled in the same way as `ManagerOptions` and other options - move the `Options` under `cmd`, rename them to e.g. `ProfilingOptions` (or `GeneralOptions`), add `Complete`, `Completed`, and `Apply` methods, and in general follow the same mechanism that is used for other options.

Reply:
Thanks @stoyanr, nice suggestion!

Reply:
I have updated the PR, PTAL.